src/sentry/models/artifactbundle.py

```diff
@@ -97,6 +111,29 @@ def delete_file_for_artifact_bundle(instance, **kwargs):
 post_delete.connect(delete_file_for_artifact_bundle, sender=ArtifactBundle)
 
 
+@region_silo_only_model
+class ArtifactBundleIndex(Model):
+    __include_in_export__ = False
+
+    organization_id = BoundedBigIntegerField(db_index=True)
+    release_name = models.CharField(max_length=250, db_index=True)
+    # We use "" in place of NULL because the uniqueness constraint doesn't play well with nullable fields, since
+    # NULL != NULL.
+    dist_name = models.CharField(max_length=64, default=NULL_STRING, db_index=True)
+    url = models.TextField(db_index=True)
+    artifact_bundle = FlexibleForeignKey("sentry.ArtifactBundle")
+    date_added = models.DateTimeField(default=timezone.now)
+    date_last_modified = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_artifactbundleindex"
+
+        index_together = (
+            ("organization_id", "release_name", "dist_name", "url", "artifact_bundle"),
```

Comment:
Q related to the indexes:

- Why do we have a non-compound indices on release/dist/url? We will never query those in isolation
- Why is `artifact_bundle` part of this compound index? We will never *search* by it *directly*. It is however a separate index implicitly by being a foreign key.
- We need a unique constraint for `org+release+dist+url`.

Reply:
1. You are right, I should also do this for our indexes in the other tables.
2. The reasoning was that we could leverage that when doing an index join on the entire composite index.

Reply:
Oh, you mean that postgres can use *only* the index to join any `artifact_bundle` table, without actually fetching the row data? Okay this would make sense I believe if this is indeed the case.

Do you still want to have a separate index for just the org_id? We don’t use it for querying *directly*, but I believe there it doesn’t hurt as much to just keep it around.

Reply:
Exactly. We can follow up with a PR to clean up indexes in the future. I have already a cleanup PR scheduled.

Reply:
> We need a unique constraint for org+release+dist+url.

Should this constraint be added now?

Reply:
Nope, we will keep the composite with also the `artifact_bundle`.

Reply:
We do need a unique constraint on it for the upsert logic to work correctly:

https://github.com/getsentry/sentry/pull/51996/files#diff-1cd089d218c5752071ba13ce89232667c385ad7c8569ba8f4f2cd4dce1774fb6R161-R183